src/bindings/g_object/type.cr

```diff
@@ -98,6 +98,19 @@ module GObject
   end
 end
 
+class Object
+  # Returns the GObject GType for pure Crystal reference objects.
+  def self.g_type
+    GObject::TYPE_POINTER
```

Comment:
Using TYPE_POINTER is bad for debugging and is thread-unsafe.

I think boxed types should be used here instead.

Reply:
boxed typed requires a boxed type to be registered in Glib type system and extra steps when we really have just a pointer to pass around.

Maybe a generic boxed type should be created to pass structs by value…. 🤔 , and use a pointer just for references.

Reply:
I added a new commit to this PR that limit is to Crystal References, so we can let structs for another time.